### PR TITLE
Add a new landing pages for client libraries & Google Cloud services

### PR DESCRIPTION
**Pull request by mit-mit**

### CODE REVIEW

**Comment by kevmoo** [comment]
I wonder if we want to DROP appengine.

I don't think we want to promote it at all. Cloud Run is the future.

CC @jonasfj

**Comment by mit-mit** [comment]
I thought AppEngine was still -- at least for now -- preferred for long-running services?

**Comment by parlough** [comment]
I agree that Cloud Run deserves a mention. There's some samples here: https://github.com/knative/docs/tree/main/community/samples/serving/helloworld-dart

**Comment by jonasfj** [comment]
Yeah, we could drop appengine.

There are still things that it's easier to do with appengine flexible.. like long running background tasks..

But our appengine flex story isn't very good in the first place. Indeed unless you're running a background task, I doubt you'll notice much difference between whether you deploy on appengine flex or cloud run.

**Comment by mit-mit** [comment]
I'll add Cloud Run. I think we should keep AppEngine.

**Comment by mit-mit** [comment]
Staged here: https://mit-staging.web.app/

* New client page: https://mit-staging.web.app/guides/services
* Cloud Run added to cloud page: https://mit-staging.web.app/server/google-cloud
* go-link added: https://mit-staging.web.app/go-cloud

File: src/server/google-cloud.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+See [GKE overview][] for more information.
+
+### App Engine
+
+Dart can be run on Google App Engine using it's _Custom Runtimes for the Flex
+Environment_ support.
+
+See the documentation in [`package:appengine`][] for more information.
+
+**Note:**: Dart's AppEngine support is incomplete and experimental. For new
```
Comment (kevmoo) [inline src/server/google-cloud.md:44]: 👍

**Review by kevmoo — APPROVED** [review]
:shipit:

**Comment by kwalrath** [comment]
I'm about to upload a massive rewrite, as is my wont. :) Not sure that I'm using the terminology correctly or that you'll agree with how I structured things... So please let me know!

The one thing I'd love to add would be a list of existing usage of each product that we want to encourage (e.g. probably not for App engine... although maybe it wouldn't be bad to mention that we have a lot of legacy usage that's been around for years). All the better if we can point to source code for real or example usage.

**Comment by kwalrath** [comment]
Staged:

https://dartlang-org-staging-1.firebaseapp.com/go/cloud (go link)
https://dartlang-org-staging-1.firebaseapp.com/server/google-cloud
https://dartlang-org-staging-1.firebaseapp.com/guides/google-apis

**Review by kevmoo — APPROVED** [review]
Love it!

**Comment by mit-mit** [comment]
@kwalrath feel free to rewrite my content any day! I really mean it, I think of what I submit as raw content meant to get us started (but I do try and learn from how I see you rewrite it).

Only one comment (see review), otherwise LGTM

File: src/server/google-cloud.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+use Google Kubernetes Engine (GKE).
+
+For more information, see the [GKE overview][].
+
+## App Engine
+
+To run Dart code on App Engine, use
+the [`appengine` package][].
+
+{{ site.alert.warn}}
```
Comment (mit-mit) [inline src/server/google-cloud.md:46]: This alert style seems very, very noisy on this page. It really draws the eyes, even for readers that had no intention of reading down to the App Engine content. 

Do we have some note style that is a bit more subtle, and only draws attention from those who are actually reading the AppEngine section?

**Comment by mit-mit** [comment]
> The one thing I'd love to add would be a list of existing usage of each product 

I agree, that would be great. For most, I think the recommendation should be Cloud Run, but we don't have that many examples yet. I'm hoping we'll get there over the next few months.

**Comment by kevmoo** [comment]
We can point folks to dartbug.com – and dartpad.dev

**Comment by kwalrath** [comment]
> We can point folks to dartbug.com – and dartpad.dev

How/where are those implemented?

File: src/server/google-cloud.md (hunk 2 above)
Comment (kwalrath) [reply src/server/google-cloud.md]: I tried rewriting it using just text, no note formatting. Check it out: https://dartlang-org-staging-1.firebaseapp.com/server/google-cloud#app-engine

**Comment by kevmoo** [comment]
> > We can point folks to dartbug.com – and dartpad.dev
> 
> How/where are those implemented?

https://github.com/dart-lang/dartbug.com

https://github.com/dart-lang/dart-services

**Comment by kwalrath** [comment]
I've added a link to the Cloud Run example (dartbug.com). I didn't add a link to the App Engine example (dart-services), since I wasn't sure it'd be helpful to people writing new servers. Also, it doesn't use the `appengine` package, so I modified that description a bit. ptal @kevmoo

File: src/server/google-cloud.md [hunk 3]
```diff
@@ -19,9 +19,14 @@ You can use Cloud Run's flexible container support,
 combined with Dart's Docker images,
 to run server-side Dart code.
 
-For more information, see the Cloud Run documentation for
+One example of a Dart server that uses Cloud Run is dartbug.com,
+the redirect service for Dart issues and bugs.
+You can find its source code in the [dart-lang/dartbug.com repo][].
```
Comment (kevmoo) [inline src/server/google-cloud.md:24]: 👍

**Review by kwalrath — APPROVED** [review]

File: src/server/google-cloud.md (hunk 2 above)
Comment (mit-mit) [reply src/server/google-cloud.md]: Nice!

**Pull request merged**